### DISCUSSION TITLE
Restart integration if the connection is lost without having to restart HA

### DISCUSSION THREAD

**Discussion by ItsMe00007** [Patrick762/hassio-bluetti-bt#75] [Q&A]
I charge my AC500+3B300S only via PV.
That causes the BMS Synchronization Problems after some time (e.g. the AC500 turns off at 20% or jumps from 95% to 100%.
Therefore i discharge it from time to time completely over night.
Therefore the AC500 is not available for the HA for a few hours and when the sun starts to charge the AC500 it is not automatically back in HA.
I have to restart the HA to get the AC500 integration back to work.

is there any other way, preferably automatically, to get the Bluetti BT integration working again without having to restart HA completely ?

**Comment by Patrick762** [comment]
Good question. I also found out that the ESPhome Bluetooth has some bugs in the latest firmware (random disconnects, no connections after restarting HA).

The big problem with this is, I don't know yet why this happens.

**Comment by Patrick762** [comment]
I'll close this because the integration changed significantly in the last few versions

**Comment by mzpwr** [comment]
@ItsMe00007 do you have "Persistent connection" set up in HA for your AC500?
I had the same problems with HA (rPI3 and integrated Bluetooth module) + EB3A + AC180P. I have WA which I found and now EB3A is automatically discovered and data from it is traceable in HA without problems and disconnections.
My current setup is the following:
1. Home Assistant rPI3 Bluetooth module. Leave default "passive scanning" as disabled; 
2. Bluetti integration leave default Bluetti device setup "Persistent connection" as disabled;
3. Try to minimize number of used BLE devices on Home Assistant node in general if you use default rPI UART Bluetooth module. I have only 2 BLE edge devices and HA works fine for both BL devices. If I add 3rd BL device, then all is messed up with tree devices and I have 3-5 minutes gaps in collected data.

I have a plan to add other USB BL adapter with CSR8510A10 (https://www.home-assistant.io/integrations/bluetooth/) to check if the eternal adapter may help with p3.

**Reply by ItsMe00007** [reply]
@mzpwr: thanks for the tip, i’ll give it a try.
Currently i have persistent connection disabled but passive scanning enabled.
I switched some time ago from using a directly attached/builtin Bluetooth connection on my Mini PC to an ESP32 Bluetooth Proxy (connected via LAN).
Since then i don‘t have any Bluetooth connection Problems.  It seems that the builtin HA Bluetooth is not very stable when using a directly connected (or a builtin) Bluetooth adapter.
